specification/botservice/resource-manager/Microsoft.BotService/stable/2017-12-01/botservice.json

```diff
@@ -392,242 +434,1239 @@
             }
           }
         }
-      }
-    },
-    "/providers/Microsoft.BotService/operations": {
-      "get": {
+      },
+      "patch": {
         "tags": [
-          "Operations"
+          "Channel"
         ],
-        "description": "Lists all the available BotService operations.",
-        "operationId": "Operations_List",
+        "description": "Updates a Channel registration for a Bot Service",
+        "operationId": "Channels_Update",
         "x-ms-examples": {
-          "Get Operations": {
-            "$ref": "./examples/GetOperations.json"
+          "Update Bot": {
+            "$ref": "./examples/UpdateChannel.json"
           }
         },
         "parameters": [
+          {
+            "name": "resourceGroupName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the resource group within the user's subscription."
+          },
+          {
+            "name": "resourceName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "maxLength": 64,
+            "minLength": 2,
+            "pattern": "^[a-zA-Z0-9][a-zA-Z0-9_.-]*$",
+            "description": "The name of the Bot resource."
+          },
+          {
+            "name": "channelName",

```

Comment:
Given that this is used multiple times, I'd again recommend refactoring it into a global parameter that's referenced in each operation for consistency's sake.

Reply:
Definitely, this will be great.